ashley/src/main/kotlin/ktx/ashley/engine/engines.kt

```diff
@@ -0,0 +1,52 @@
+package ktx.ashley.engine
+
+import com.badlogic.ashley.core.Component
+import com.badlogic.ashley.core.Engine
+import com.badlogic.ashley.core.Entity
+import ktx.ashley.AshleyDsl
+
+/**
+ * Creates an instance of the component [T] and adds it to this [Entity].
+ *
+ * @param T the [Component] type to get or create
+ * @param configure inlined function with [T] as the receiver to allow additional configuration of the [Component]
+ * @return the created [Component]
+ */
+inline fun <reified T: Component> Entity.with(configure: (@AshleyDsl T).() -> Unit): T {
+  return with<T>().also(configure)
+}
+
+/**
+ * Creates a pooled instance of the component [T] and adds it to this [Entity].
+ *
+ * @param T the [Component] type to get or create
+ * @return the pooled [Component]
+ */
+inline fun <reified T: Component> Entity.with(): T {
+  val component = T::class.java.newInstance()
```

Comment:
I just noticed that `PooledEngine` extension methods would prevent reflection usage. Well, if you _want_ to refactor the code not to use reflection, I agree. If you think `PooledEngine` will usually be used most of the time, you can use [LibGDX reflection API](https://github.com/libgdx/libgdx/wiki/Reflection) instead. Add a warning (in doc comment) that it uses reflection, though.